org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/launch/npm/NPMLaunchTab.java

```diff
@@ -0,0 +1,172 @@
+
+		Composite emptySpace = new Composite(resComposite, SWT.NONE);
+		emptySpace.setLayoutData(new GridData(SWT.FILL, SWT.DEFAULT, false, false));
+
+		Composite fileComposite = new Composite(resComposite, SWT.NONE);
+		fileComposite.setLayout(new GridLayout(2, false));
+		fileComposite.setLayoutData(new GridData(SWT.FILL, SWT.DEFAULT, true, false));
+		new Label(fileComposite, SWT.NONE).setText("package.json file:");
+		this.programPathText = new Text(fileComposite, SWT.BORDER);
+		this.programPathText.setLayoutData(new GridData(SWT.FILL, SWT.DEFAULT, true, false));
+		ControlDecoration decoration = new ControlDecoration(programPathText, SWT.TOP | SWT.LEFT);
+		FieldDecoration fieldDecoration = FieldDecorationRegistry.getDefault()
+				.getFieldDecoration(FieldDecorationRegistry.DEC_ERROR);
+		decoration.setImage(fieldDecoration.getImage());
+		this.programPathText.addModifyListener(event -> {
+			setDirty(true);
+			File file = new File(programPathText.getText());
+			if (!file.isFile()) {
+				String errorMessage = Messages.RunProgramTab_error_unknownFile;
+				setErrorMessage(errorMessage);
+				decoration.setDescriptionText(errorMessage);
+				decoration.show();
+			} else if (!shortcut.canLaunch(file)) {
+				String errorMessage = "Not a package.json file";
+				setErrorMessage(errorMessage);
+				decoration.setDescriptionText(errorMessage);
+				decoration.show();
+			} else if (!file.canRead()) {
+				String errorMessage = Messages.RunProgramTab_error_nonReadableFile;
+				setErrorMessage(errorMessage);
+				decoration.setDescriptionText(errorMessage);
+				decoration.show();
+			} else {
+				setErrorMessage(null);
+				decoration.hide();
+			}
+			updateLaunchConfigurationDialog();
+		});
+
+		Button filePath = new Button(resComposite, SWT.PUSH);
+		filePath.setText("Browse...");
+		filePath.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
+			FileDialog filePathDialog = new FileDialog(resComposite.getShell());
+			filePathDialog.setFilterPath(getSelectedProject() == null ? null : getSelectedProject().getAbsolutePath());
+			filePathDialog.setText("Select a package.json to use");
+			String path = filePathDialog.open();
+			if (path != null) {
+				packageJSONFile = new File(path);
+				programPathText.setText(packageJSONFile.getAbsolutePath());
+				setDirty(true);
+				updateLaunchConfigurationDialog();
+			}
+		}));
+
+		setControl(resComposite);
+	}
+
+	@Override
+	public void setDefaults(ILaunchConfigurationWorkingCopy configuration) {
+		// Nothing to do
+	}
+
+	@Override
+	public void initializeFrom(ILaunchConfiguration configuration) {
+		try {
+			defaultSelectedFile = getSelectedFile(shortcut::canLaunch);
+			String defaultSelectedFilePath = pathOrEmpty(defaultSelectedFile);
+			this.programPathText
+					.setText(configuration.getAttribute(AbstractHTMLDebugDelegate.PROGRAM, defaultSelectedFilePath));
+			this.argumentsCombo.setText(configuration.getAttribute(AbstractHTMLDebugDelegate.ARGUMENTS, "install"));
```

Comment:
externalize `"install"`

Reply:
no need to externalize here, this is not a translatable text.

Reply:
good call thanks